src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -395,5 +361,48 @@ private BoundStatement RewriteUsingStatementTryFinally(SyntaxNode syntax, BoundB
 
             return tryFinally;
         }
+
+        private BoundExpression GenerateDisposeCall(SyntaxNode syntax, BoundExpression disposedExpression, MethodSymbol methodOpt, AwaitableInfo awaitOpt, SyntaxToken awaitKeyword)
+        {
+            // If we don't have an explicit dispose method, try and get the special member for IDiposable/IAsyncDisposable
+            if (methodOpt is null)
+            {
+                if (awaitOpt is null)
+                {
+                    // IDisposable.Dispose()
+                    Binder.TryGetSpecialTypeMember(_compilation, SpecialMember.System_IDisposable__Dispose, syntax, _diagnostics, out methodOpt);
+                }
+                else
+                {
+                    Debug.Assert(awaitKeyword != default);
```

Comment:
This assert is now only checked when `methodOpt is null && !(awaitOpt is null)`. This potentially misses cases where `await` information was passed down and there happened to be a `methodOpt`. Could we move the assert up and change it to: `Debug.Assert(awaitOpt is null || awaitKeyword != default)`

Reply:
Good catch. Your fix makes sense. Fixed.